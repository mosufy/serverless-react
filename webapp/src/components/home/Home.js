import React, { Component } from 'react';

import Featurette from "./Featurette";
import Hero from "./Hero";

import Featurette1Img from "../../assets/img/featurette1.jpg";
import Featurette2Img from "../../assets/img/featurette2.jpg";
import Featurette3Img from "../../assets/img/featurette3.jpg";
import RegisterInterestModal from "../modal/RegisterInterestModal";

class Home extends Component {
  render() {
    let { showRegisterInterestModal, handleRegisterInterestModalOnClick, handleRegisterInterestOnSubmit } = this.props;

    return (
      <div>
        <Hero/>

        <div className="container">
          <hr className="featurette-divider"/>

          <div id="about"/>

          <Featurette imgSrc={Featurette1Img}
                      pullDirection="right"
                      heading="One-On-One"
                      subHeading="Because Everyone Is Different"
                      text="Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo."
                      handleButtonClick={handleRegisterInterestModalOnClick}/>

          <hr className="featurette-divider"/>

          <Featurette imgSrc={Featurette2Img}
                      pullDirection="left"
                      heading="MOE Certified Tutors"
                      subHeading="Are Our Quality Assurance"
                      text="Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo."
                      handleButtonClick={handleRegisterInterestModalOnClick}/>

          <hr className="featurette-divider"/>

          <Featurette imgSrc={Featurette3Img}
                      pullDirection="right"
                      heading="This Third Heading"
                      subHeading="Will Seal the Deal"
                      text="Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo."
                      handleButtonClick={handleRegisterInterestModalOnClick}/>

          <hr className="featurette-divider"/>

          <RegisterInterestModal showModal={showRegisterInterestModal}
                                 handleModalOnClick={handleRegisterInterestModalOnClick}
                                 handleRegisterInterestOnSubmit={handleRegisterInterestOnSubmit}/>

        </div>
      </div>
    );
  }
}

export default Home;