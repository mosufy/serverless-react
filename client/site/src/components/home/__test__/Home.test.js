import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../Home';

describe('home.Home component testing', () => {
  it('it renders correctly', () => {
    const tree = renderer.create(
      <Home />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});