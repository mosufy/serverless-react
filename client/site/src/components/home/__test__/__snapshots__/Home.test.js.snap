// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.Home component testing it renders correctly 1`] = `
<div>
  <header
    className="header-image"
  >
    <div
      className="headline"
    >
      <div
        className="container"
      >
        <h1>
          Sher Tuition
        </h1>
        <h2>
          Improving Life Together
        </h2>
      </div>
    </div>
  </header>
  <div
    className="container"
  >
    <hr
      className="featurette-divider"
    />
    <div
      id="about"
    />
    <div
      className="featurette"
    >
      <img
        alt=""
        className="featurette-image img-circle img-responsive pull-right"
        src="featurette1.jpg"
      />
      <h2
        className="featurette-heading"
      >
        One-On-One
         
        <span
          className="text-muted"
        >
          Because Everyone Is Different
        </span>
      </h2>
      <p
        className="lead"
      >
        Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo.
      </p>
      <button
        className="btn btn-lg btn-danger"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Register Interest
      </button>
    </div>
    <hr
      className="featurette-divider"
    />
    <div
      className="featurette"
    >
      <img
        alt=""
        className="featurette-image img-circle img-responsive pull-left"
        src="featurette2.jpg"
      />
      <h2
        className="featurette-heading"
      >
        MOE Certified Tutors
         
        <span
          className="text-muted"
        >
          Are Our Quality Assurance
        </span>
      </h2>
      <p
        className="lead"
      >
        Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo.
      </p>
      <button
        className="btn btn-lg btn-danger"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Register Interest
      </button>
    </div>
    <hr
      className="featurette-divider"
    />
    <div
      className="featurette"
    >
      <img
        alt=""
        className="featurette-image img-circle img-responsive pull-right"
        src="featurette3.jpg"
      />
      <h2
        className="featurette-heading"
      >
        This Third Heading
         
        <span
          className="text-muted"
        >
          Will Seal the Deal
        </span>
      </h2>
      <p
        className="lead"
      >
        Donec ullamcorper nulla non metus auctor fringilla. Vestibulum id ligula porta felis euismod semper. Praesent commodo cursus magna, vel scelerisque nisl consectetur. Fusce dapibus, tellus ac cursus commodo.
      </p>
      <button
        className="btn btn-lg btn-danger"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Register Interest
      </button>
    </div>
    <hr
      className="featurette-divider"
    />
  </div>
</div>
`;
